types.go

```diff
@@ -15,33 +15,34 @@ import (
 
 const defaultGradientAbsTol = 1e-6
 
-// EvaluationType is used by the optimizer to specify information needed
-// from the objective function.
+// EvaluationType is used by the optimizer to specify information needed from
+// the objective function. The types can be composed together using binary or
+// operator.
 type EvaluationType int
 
 const (
-	NoEvaluation EvaluationType = iota
-	FuncEvaluation
-	GradEvaluation
-	HessEvaluation
-	FuncGradEvaluation
-	FuncGradHessEvaluation
+	NoEvaluation           EvaluationType = 0
```

Comment:
Ah, I understand the bitwise-and now. Still, we should be able to return using "+" rather than using the bitwise or, right?